mint-auditor/src/gnosis/eth_data_types.rs

```diff
@@ -58,18 +101,29 @@ impl FromStr for EthTxHash {
             return Err(Error::InvalidTxHash(src.to_string()));
         }
 
-        let bytes = hex::decode(&src[2..]).map_err(|_| Error::InvalidTxHash(src.to_string()))?;
-        if bytes.len() != 32 {
-            return Err(Error::InvalidTxHash(src.to_string()));
-        }
-
-        Ok(Self(src.to_string()))
+        Self::try_from(
+            &hex::decode(&src[2..]).map_err(|_| Error::InvalidTxHash(src.to_string()))?[..],
+        )
     }
 }
 
 impl fmt::Display for EthTxHash {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-        write!(f, "{}", self.0)
+        write!(f, "0x{}", hex::encode(&self.0))
+    }
+}
+
+impl AsRef<[u8]> for EthTxHash {
+    fn as_ref(&self) -> &[u8] {
+        &self.0
+    }
+}
+
+impl FromRandom for EthTxHash {
+    fn from_random<T: RngCore + CryptoRng>(rng: &mut T) -> Self {
+        let mut result = Self([0u8; Self::LEN]);
+        rng.fill_bytes(&mut result.0);
+        result
```

Comment:
Optional/Equivalent: Use the existing impl for byte arrays.
```suggestion
        Self(FromRandom::from_random(rng))
```